toolchain/lower/mangler.cpp

```diff
@@ -0,0 +1,99 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "toolchain/lower/mangler.h"
+
+#include "toolchain/base/kind_switch.h"
+#include "toolchain/sem_ir/entry_point.h"
+
+namespace Carbon::Lower {
+
+auto Mangler::MangleInverseQualifiedNameScope(llvm::raw_ostream& os,
+                                              SemIR::NameScopeId name_scope_id)
+    -> void {
+  struct NameEntry {
+    SemIR::NameScopeId name_id;
```

Comment:
```suggestion
    SemIR::NameScopeId name_scope_id;
```

We do also have a `NameId` structure -- I think we tend to call these either `scope_id` or `name_scope_id`